test/suites/integration/ram_share_test.go

```diff
@@ -0,0 +1,352 @@
+		stsClient := sts.New(primarySess)
+		assumeRoleInput := &sts.AssumeRoleInput{
+			RoleArn:         aws.String(SecondaryAccountTestRoleArn),
+			RoleSessionName: aws.String("aws-application-networking-k8s-ram-e2e-test"),
+		}
+
+		assumeRoleResult, err := stsClient.AssumeRoleWithContext(ctx, assumeRoleInput)
+		Expect(err).NotTo(HaveOccurred())
+
+		creds := assumeRoleResult.Credentials
+
+		secondarySess := session.Must(session.NewSession(&aws.Config{
+			Region: aws.String(config.Region),
+			Credentials: credentials.NewStaticCredentials(
+				*creds.AccessKeyId,
+				*creds.SecretAccessKey,
+				*creds.SessionToken,
+			),
+		}))
+
+		primaryRamClient = ram.New(primarySess)
+		secondaryRamClient = ram.New(secondarySess)
+		primaryVpcLatticeClient = vpclattice.New(primarySess)
+		secondaryVpcLatticeClient = vpclattice.New(secondarySess)
+		primaryResourceGroupsClient = resourcegroupstaggingapi.New(primarySess)
+		secondaryResourceGroupsClient = resourcegroupstaggingapi.New(secondarySess)
+	})
+
+	It("makes service network from secondary account usable for gateway with matching name", func() {
+		randomName := k8sResourceNamePrefix + utils.RandomAlphaString(10)
+		createSharedServiceNetwork(
+			randomName,
+			secondaryAccountId,
+			primaryVpcLatticeClient,
+			secondaryVpcLatticeClient,
+			primaryRamClient,
+			secondaryRamClient,
+		)
+
+		// Create gateway with same name as service network
+		gateway := &gwv1.Gateway{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      randomName,
+				Namespace: k8snamespace,
+			},
+			Spec: gwv1.GatewaySpec{
+				GatewayClassName: "amazon-vpc-lattice",
+				Listeners: []gwv1.Listener{
+					{
+						Name:     gwv1.SectionName(randomName),
+						Port:     gwv1.PortNumber(80),
+						Protocol: gwv1.HTTPProtocolType,
+					},
+				},
+			},
+		}
+		err := testFramework.Create(ctx, gateway)
+		Expect(err).ToNot(HaveOccurred())
+
+		Eventually(func(g Gomega) {
+			// Policy status should be Accepted
+			gwNamespacedName := types.NamespacedName{
+				Name:      gateway.Name,
+				Namespace: gateway.Namespace,
+			}
+			gw := &gwv1.Gateway{}
+			err := testFramework.Client.Get(ctx, gwNamespacedName, gw)
+			g.Expect(err).To(BeNil())
+			g.Expect(len(gw.Status.Conditions) >= 2).To(BeTrue())
+			g.Expect(gw.Status.Conditions[1].Type).To(BeEquivalentTo(string(gwv1.GatewayConditionProgrammed)))
+			g.Expect(gw.Status.Conditions[1].Status).To(BeEquivalentTo(metav1.ConditionTrue))
+			g.Expect(gw.Status.Conditions[1].Reason).To(BeEquivalentTo(string(gwv1.GatewayReasonProgrammed)))
+		}).Should(Succeed())
+	})
+
+	It("makes service network from secondary account usable for gateway with matching id", func() {
+		randomName := k8sResourceNamePrefix + utils.RandomAlphaString(10)
+
+		serviceNetwork := createSharedServiceNetwork(
+			randomName,
+			secondaryAccountId,
+			primaryVpcLatticeClient,
+			secondaryVpcLatticeClient,
+			primaryRamClient,
+			secondaryRamClient,
+		)
```

Comment:
since you are using same setup, you can create shared network once for all tests in BeforeAll

Reply:
Name of service network depends specifically on the test. Will leave it as a function.